lib/fluent/plugin/out_elasticsearch.rb

```diff
@@ -349,12 +357,95 @@ def get_parent_of(record, path)
     [parent_object, path[-1]]
   end
 
-  def send_bulk(data)
+  def send_bulk(data, count)
     retries = 0
     begin
       response = client.bulk body: data
       if response['errors']
-        log.error "Could not push log to Elasticsearch: #{response}"
+        errors = Hash.new(0)
```

Comment:
Any chance to extract into method this _too_ long error classifying implementation?
This implementation is not easy to be testable.

Reply:
would you like everything in the `if response['errors']` block to be in a separate method, or would you like it broken down more?

Reply:
I want to wrap these lines into one or a few encapsulated method.

Reply:
Or, extract `HandleError` class.